server/routes/update-goal/UpdateGoalController.test.ts

```diff
@@ -117,6 +117,16 @@ describe('UpdateGoalController', () => {
       expect(next).toHaveBeenCalledWith(new Error('different steps were submitted'))
     })
 
+    it('should redirect to the plan page when saved with no steps and no note', async () => {
```

Comment:
I'm not sure this is right - should it not throw a validation error asking for something to be added? Can we check with UX?